### PR TITLE
use grafana 5 datasource and dashboard provisioning

### PR DESCRIPTION
**Pull request by DaytonG**
With Grafana 5 we have newer, somewhat simpler, methods to provision datasources and dashboards via config files. That's my main goal for this PR.

Secondarily:
- I changed the values.yaml to allow easier overrides for the image registry (my company always pulls images from a private registry)
- simplified variables and defaults. Instead of using the `default` template function, I allow the values.yaml to _be_ the defaults, thereby simplifying some code and reducing missing variable errors.

I realize this is a pretty massive PR, so feel free to rip it to shreds!

Edit: for details on the new provisioning config: http://docs.grafana.org/administration/provisioning/

### CODE REVIEW

File: stable/grafana/README.md
Comment (timstoop) [inline stable/grafana/README.md]: I'm not necessarily against this, but you're aware it's fairly uncommon to do this, right? In general it's agreed that `repository` should be the entire name of the image. At least in the Charts I tend to use.

File: stable/grafana/README.md
Comment (timstoop) [inline stable/grafana/README.md]: Please remove the `(AKA version)` part, it's unnecessary.

File: stable/grafana/README.md
Comment (timstoop) [inline stable/grafana/README.md]: Please consider lining up the | for the entire table. I know it seems a bit silly, but it makes editing and reading the source a lot more pleasant. The next committer will thank you for it!

**Review by timstoop — CHANGES_REQUESTED** [review]
Out of curiosity, do the datasources contain anything that warrants them to be a Secret? Which kind of sensitive information would be in there?

File: stable/grafana/README.md
Comment (DaytonG) [reply stable/grafana/README.md]: Yea, I'm fine with doing it either way. The charts I've used have followed the pattern you've experienced. That being said, at least for my use case, I've found it rather nice to have the "registry" portion in its own variable. It makes overriding that one portion the smallest possible change, with my overrides not even caring about the actual "name" of the image.

**Comment by DaytonG** [comment]
@timstoop The datasources configuration can feasibly contain usernames, passwords, TLS certificate+keys, and other authentication options.

**Comment by DaytonG** [comment]
While doing some further local testing, I noticed that my use case (a private image mirror), I need to be able to set a pull secret. So I added that.

**Comment by rtluckie** [comment]
please check out #4713 before merging

**Comment by DaytonG** [comment]
closing in favor of #4713

**Pull request closed without merging**
